src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/SubscriptionsController.cs

```diff
@@ -109,31 +109,49 @@ public virtual async Task<IActionResult> GetSubscription(Guid id)
         ///   Trigger a <see cref="Subscription"/> manually by id
         /// </summary>
         /// <param name="id">The id of the <see cref="Subscription"/> to trigger.</param>
+        /// <param name="buildId">'bar-build-id' if specified, a specific build is requested</param>
         [HttpPost("{id}/trigger")]
         [SwaggerApiResponse(HttpStatusCode.Accepted, Type = typeof(Subscription), Description = "Subscription update has been triggered")]
         [ValidateModelState]
-        public virtual async Task<IActionResult> TriggerSubscription(Guid id)
+        public virtual async Task<IActionResult> TriggerSubscription(Guid id, [FromQuery(Name = "bar-build-id")] int buildId = 0)
         {
-            Data.Models.Subscription subscription = await TriggerSubscriptionCore(id);
+            Data.Models.Subscription subscription = await TriggerSubscriptionCore(id, buildId);
 
             if (subscription == null)
                 return NotFound();
 
             return Accepted(new Subscription(subscription));
         }
 
-        protected async Task<Data.Models.Subscription> TriggerSubscriptionCore(Guid id)
+        protected async Task<Data.Models.Subscription> TriggerSubscriptionCore(Guid id, int buildId = 0)
         {
             Data.Models.Subscription subscription = await _context.Subscriptions.Include(sub => sub.LastAppliedBuild)
                 .Include(sub => sub.Channel)
                 .FirstOrDefaultAsync(sub => sub.Id == id);
 
+            if (buildId != 0)
+            {
+                Data.Models.Build build = await _context.Builds.Where(b => b.Id == buildId).FirstOrDefaultAsync();
+                // Non-existent build
+                if (build == null)
+                {
+                    return null;
+                }
+                // Build doesn't match source repo
+                if (!(build.GitHubRepository?.Equals(subscription.SourceRepository, StringComparison.InvariantCultureIgnoreCase) == true ||
+                      build.AzureDevOpsRepository?.Equals(subscription.SourceRepository, StringComparison.InvariantCultureIgnoreCase) == true))
+                {
+                    return null;
```

Comment:
Do you think we should return BadRequest in this case?

Reply:
Sure; null was the other default error case so I was mostly following the pattern, seems reasonable to make it fail more loudly.